src/python/pants/backend/go/target_type_rules_test.py

```diff
@@ -139,6 +146,6 @@ def test_go_package_dependency_inference(rule_runner: RuleRunner) -> None:
         InferredDependencies, [InferGoPackageDependenciesRequest(target2[GoPackageSources])]
     )
     assert inferred_deps_2.dependencies == FrozenOrderedSet(
-        [Address("foo", target_name="github.com_google_go-cmp_0.4.0")]
+        [Address("foo", target_name="github.com_google_go-cmp_0.4.0-_cmp")]
```

Comment:
This looks weird. Should you be stripping the leading `/` from the subpath rather than converting to `_`?